app/src/androidTest/java/com/github/geohunt/app/ui/components/profile/edit/ProfileEditPageTest.kt

```diff
@@ -20,46 +17,68 @@ class ProfileEditPageTest {
     @get:Rule
     val c = createComposeRule()
 
-    @Before
-    fun mockAuthenticator() {
-        Authenticator.authInstance.set(LoginActivityTest.MockAuthenticator(MockUser(uid = "1")))
+    private fun createViewModel(authRepository: MockAuthRepository = MockAuthRepository(), userRepository: MockUserRepository = MockUserRepository()): ProfileEditPageViewModel {
+        return ProfileEditPageViewModel(
+            authRepository,
+            userRepository
+        )
     }
 
     @Test
-    fun showsLoadingIfNotAvailableYet() {
-        val mockDb = object: BaseMockDatabase() {
-            override fun getUserById(uid: String): LiveLazyRef<User> {
-                return InstantLazyRef("1", null)
+    fun showsLoadingIfNotReadyYet() {
+        val cf = CompletableFuture<Void?>()
+        val vm = createViewModel(
+            userRepository = object: MockUserRepository() {
+                override suspend fun getCurrentUser(): User {
+                    return withContext(Dispatchers.IO) {
+                        cf.get()
+                        MockAuthRepository.defaultLoggedUser
+                    }
```

Comment:
Avoid using `Dispatchers.IO` in tests, as the test might become flaky, and it does not benefit for auto-advancing virtual time. You can use the [test dispatchers](https://developer.android.com/kotlin/coroutines/test#testdispatchers) for that purpose.